README.md

```diff
@@ -1,7 +1,25 @@
-# Fabric
+# Fabric API
 
 Essential hooks and patches for modding with Fabric.
 
+Fabric API is the library for essential hooks and interop mechanisms for Fabric mods. Examples include:
+
+- Exposing functionality that is useful but difficult to access for many mods such as particles, biomes and dimensions
+- Adding events, hooks and APIs to improve interopability between mods.
+- Essential features such as registry synchronization and adding information to crash reports.
+- An advanced rendering API designed for compatibility with optimization mods and graphics overhaul mods.
+
+Also check out [Fabric Loader](https://github.com/FabricMC/fabric-loader), the (mostly) version independent mod loader that powers Fabric. Fabric API is a mod like any other which requires Fabric Loader to be installed.
+
+For support and discussion for both developers and users, visit [the Fabric Discord server](https://discord.gg/v6v4pMv).
+
 ## Getting Started
 
 To setup a Fabric development environment, check out the [example fabric mod](https://github.com/FabricMC/fabric-example-mod) and follow the instructions there.
+
+## Contributing
+
+[Checkstyle](https://checkstyle.sourceforge.io/) is used to enforce code style. Make sure to install the necessary tools in your development environment to not be held up in review by failing code style checks.
+
+For more information on contributing to Fabric, read [The Fabric Feature Procedure](https://fabricmc.net/wiki/tutorial:feature_procedure).
```

Comment:
Honestly I feel going full out with a `CONTRIBUTING.MD` file would suit this much better.